packages/core/src/version-utils.test.ts

```diff
@@ -0,0 +1,268 @@
+    const fetchSpy = jest.spyOn(globalThis, 'fetch').mockImplementation(
+      // @ts-expect-error Not perfect match to fetch
+      jest.fn(async () => {
+        return Promise.resolve({
+          status: 200,
+          json: async () => {
+            return manifest;
+          },
+        });
+      })
+    );
+    await expect(fetchVersionManifest('3.1.6')).resolves.toMatchObject<ReleaseManifest>(manifest);
+    // Should be called with latest
+    expect(fetchSpy).toHaveBeenLastCalledWith(`${GITHUB_RELEASES_URL}/tags/v3.1.6`);
+    // Call again to make sure we don't refetch
+    fetchSpy.mockClear();
+    await expect(fetchVersionManifest('3.1.6')).resolves.toMatchObject<ReleaseManifest>(manifest);
+    expect(fetchSpy).not.toHaveBeenCalled();
+    fetchSpy.mockRestore();
+  });
+
+  test('Fetch throws -- Network error', async () => {
+    const fetchSpy = jest.spyOn(globalThis, 'fetch').mockImplementation(
+      jest.fn(async () => {
+        return Promise.reject(new Error('Network request failed'));
+      })
+    );
+    await expect(fetchVersionManifest('3.1.6')).rejects.toThrow('Network request failed');
+    fetchSpy.mockRestore();
+  });
+
+  test('Version not found', async () => {
+    const fetchSpy = jest.spyOn(globalThis, 'fetch').mockImplementation(
+      // @ts-expect-error Not perfect match to fetch
+      jest.fn(async () => {
+        return Promise.resolve({
+          status: 404,
+          json: async () => {
+            return { message: 'Not Found' };
+          },
+        });
+      })
+    );
+    await expect(fetchVersionManifest('3.1.6')).rejects.toThrow(
+      "Received status code 404 while fetching manifest for version '3.1.6'. Message: Not Found"
+    );
+    fetchSpy.mockRestore();
+  });
+});
+
+describe('fetchLatestVersionString', () => {
+  beforeEach(() => {
+    clearReleaseCache();
+  });
+
+  test('Successful', async () => {
+    const manifest = {
+      tag_name: 'v3.1.6',
+      assets: [
+        {
+          name: 'medplum-agent-3.1.6-linux',
+          browser_download_url: 'https://example.com',
+        },
+      ],
+    } as ReleaseManifest;
+    const fetchSpy = jest.spyOn(globalThis, 'fetch').mockImplementation(
+      // @ts-expect-error Not perfect match to fetch
+      jest.fn(async () => {
+        return Promise.resolve({
+          status: 200,
+          json: async () => {
+            return manifest;
+          },
+        });
+      })
+    );
+    await expect(fetchLatestVersionString()).resolves.toEqual('3.1.6');
+    fetchSpy.mockRestore();
+  });
+
+  test('Invalid latest release', async () => {
+    const manifest = {
+      tag_name: 'canary',
+      assets: [
+        {
+          name: 'medplum-agent-canary-linux',
+          browser_download_url: 'https://example.com',
+        },
+      ],
+    } as ReleaseManifest;
+    const fetchSpy = jest.spyOn(globalThis, 'fetch').mockImplementation(
+      // @ts-expect-error Not perfect match to fetch
```

Comment:
Can we cast rather than use `@ts-expect-error`? 

`@ts-expect-error` should be used for exceptional cases, because it can hide other valid errors.

Reply:
I kinda agree but also like that there is a lint rule for explaining why this `expect-error` is necessary (kinda reminds me of Rust unsafe), though might not be a good enough reason to use it